### PR TITLE
Port encrypted config shared bits to a separate PR

### PR DESCRIPTION
**Pull request by jefferai**
See #8562 for more information about this. This simply ports shared parts over to a separate PR.

### CODE REVIEW

File: internalshared/configutil/encrypt_decrypt.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"regexp"
+
+	"github.com/gogo/protobuf/proto"
```
Comment (briankassouf) [inline internalshared/configutil/encrypt_decrypt.go:11]: Did you mean to use gogo here?

File: internalshared/configutil/encrypt_decrypt_test.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	circonus_api_key = "barfoo"
+}
+`
+
+	reverser := new(reversingWrapper)
+	out, err := EncryptDecrypt(rawStr, false, false, reverser)
+	if err != nil {
+		t.Fatal(err)
+	}
+
```
Comment (briankassouf) [inline internalshared/configutil/encrypt_decrypt_test.go:41]: Should you check the encrypted output contains `{{decrypt(oofrab)}}`?

File: internalshared/configutil/encrypt_decrypt.go (hunk 1 above)
Comment (jefferai) [reply internalshared/configutil/encrypt_decrypt.go:11]: Sure didn't. That stuff is insidious.  Will fix.

File: internalshared/configutil/encrypt_decrypt_test.go (hunk 2 above)
Comment (jefferai) [reply internalshared/configutil/encrypt_decrypt_test.go:41]: I actually was thinking about that -- technically it contains a marshaled proto but I can just unmarshal in the test and verify. I sort of figured so long as what came in == what goes out it doesn't truly matter what it looks like in the middle 😂  But I'll add it.

**Pull request merged**